imap/jmap_util.h

```diff
@@ -147,4 +148,24 @@ extern int jmap_decode_rawdata_blobid(const char *blobid,
                                       char **subpartptr,
                                       struct message_guid *guid);
 
+enum header_form {
+    HEADER_FORM_UNKNOWN          = 0, /* MUST be zero so we can cast to void* */
+    HEADER_FORM_RAW              = 1 << 0,
+    HEADER_FORM_TEXT             = 1 << 1,
+    HEADER_FORM_DATE             = 1 << 2,
+    HEADER_FORM_URLS             = 1 << 3,
+    HEADER_FORM_MESSAGEIDS       = 1 << 4,
+    HEADER_FORM_ADDRESSES        = 1 << 5,
+    HEADER_FORM_GROUPEDADDRESSES = 1 << 6
+};
+
+extern json_t *jmap_header_as_raw(const char *raw, enum header_form form);
```

Comment:
Now that we export these `as_xxx` functions, I think we should remove the `form` argument. All of these functions but the `as_addresses` and `from_addr` functions ignore them, and the reason they all accept an unused argument is because we internally use a hash table of converter functions in `jmap_mail.c`. I wonder if we can just split the `as_addresses` in two functions `as_addresses` and `as_groupaddresses` and do away with that `form` argument.